src/Diagnostics.RuntimeHost/Services/StorageService/StorageService.cs

```diff
@@ -126,11 +127,12 @@ public async Task<List<DiagEntity>> GetEntitiesByPartitionkey(string partitionKe
                     } while (tableContinuationToken != null);
                     timeTakenStopWatch.Stop();
                     DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageService), $"GetEntities by Partition key {partitionKey} took {timeTakenStopWatch.ElapsedMilliseconds}, Total rows = {detectorsResult.Count}, ClientRequestId = {clientRequestId} ");
-                    return detectorsResult.Where(result => !result.IsDisabled).ToList();
+                     return startTime == DateTime.MinValue ? detectorsResult.Where(result => !result.IsDisabled).ToList() :
+                        detectorsResult.ToList();
```

Comment:
Just making sure I understand this correctly, you want to return detectors regardless if they were disabled or not that were created/updated in the last 5 mins?

https://github.com/Azure/Azure-AppServices-Diagnostics/blob/df3b71f148ae7e0021a2ac6b1f3e44c29754bfdc/src/Diagnostics.RuntimeHost/Services/CacheService/DiagEntityTableCacheService.cs#L45

https://github.com/Azure/Azure-AppServices-Diagnostics/blob/df3b71f148ae7e0021a2ac6b1f3e44c29754bfdc/src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs#L191-L195

Reply:
Yes that's correct.  This is because people might delete detectors through PR and when we refresh our in-memory cache, we have to get all detectors regardless they were disabled or not within last 5 mins and remove the ones that were deleted.
Currently there is a limitation that deleted detectors won't go away in the UI until the process is restarted.